tests/integration/components/dialog-test.js

```diff
@@ -499,4 +499,29 @@ module('Integration | Component | <Dialog>', function (hooks) {
       async function () {}
     );
   });
+
+  module('InitialFocus', function() {
+    test('it should focus requested element', async function() {
+      this.set('isOpen', false);
+      let buttonElement = () => getByText('focused');
+
+      await render(hbs`
+        <button type="button" {{on "click" (set this.isOpen true)}}>
+          Trigger
+        </button>
+        <Dialog
+          @isOpen={{this.isOpen}}
+          @onClose={{set this.isOpen false}}
+          @initialFocus={{this.button}}
+        >
+          <input type="text" value="first">
+          <button type="button">focused</button>
+        </Dialog>
+      `);
+      this.set('button', buttonElement);
```

Comment:
Oh, I see....you're using the fact that focus-trap `initialFocus` can accept a function that returns an HTMLElement. I wonder whether for simplicity of the test, we should just us a css selector, maybe something like this:

```hbs
<button type="button" {{on "click" (set this.isOpen true)}}>
  Trigger
</button>
<Dialog
  @isOpen={{this.isOpen}}
  @onClose={{set this.isOpen false}}
  @initialFocus="[data-test-focussed-button]"
>
  <input type="text" value="first">
  <button data-test-focussed-button type="button">focused</button>
</Dialog>
```

That way we can remove the `buttonElement` function altogether.